include/matx/operators/isclose.h

```diff
@@ -0,0 +1,109 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2021, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/scalar_ops.h"
+#include "matx/operators/base_operator.h"
+
+namespace matx
+{
+
+  namespace detail {
+    template <typename Op1, typename Op2> 
+    class IsCloseOp : public BaseOp<IsCloseOp<Op1, Op2>>
+    {
+      public:
+        using matxop = bool;
+        using scalar_type = typename remove_cvref_t<Op2>::scalar_type;
+        using inner_type = typename inner_op_type_t<scalar_type>::type;
+
+        __MATX_INLINE__ std::string str() const { return "isclose()"; }
+
+        __MATX_INLINE__ IsCloseOp(Op1 op1, Op2 op2, double rtol, double atol) : 
+          op1_(op1), op2_(op2), rtol_(static_cast<inner_type>(rtol)), atol_(static_cast<inner_type>(atol)) 
+        {
+          static_assert(op1.Rank() == op2.Rank(), "Operator ranks must match in isclose()");
```

Comment:
Can you use  ASSERT_COMPATIBLE_OP_SIZES(op1);   ASSERT_COMPATIBLE_OP_SIZES(op2); here?

Reply:
Is there any reason the sizes must be equal?  This seems to prohibit broadcasting.   For example I could see someone doing something like this  isclose(op, 1).  ASSERT_COMPATIBILE_OP_SIZES is what you want I think.

Reply:
Removed the restriction